docs/modules/ROOT/pages/pallets/pallet_xcm.adoc

```diff
@@ -0,0 +1,340 @@
+:source-highlighter: highlight.js
+:highlightjs-languages: rust
+:github-icon: pass:[<svg class="icon"><use href="#github-icon"/></svg>]
+
+= pallet_xcm
+
+Branch/Release: `release-polkadot-v{1.3.0}`
+
+== Purpose
+
+`pallet-xcm` is responsible for filtering, routing, and executing incoming XCM.
+
+== Glossary
+
+** `Barrier` -- whether to execute the XCM call at all
+** `Fungible Asset` -- any unit is worth the same
+** `NonFungible Asset` -- each unit is worth a different amount and/or is unique in some way
+** `Junction` -- single item in a path to describe a relative location (think of `dir` or `dir/$file_path` as a junction)
+** `Junctions` -- multiple `Junction`s, formed only through MultiLocation construction
+** `MultiAsset` -- Either an amount of a single fungible asset, or a single well-identified non-fungible asset.
+** `MultiLocation` -- path described by junctions leading to the location
+** `MAX_ASSETS_FOR_TRANSFER` -- The maximum number of distinct assets allowed to be transferred in a single helper extrinsic. Set to 2 in this code.
+
+== Config link:https://github.com/paritytech/polkadot-sdk/blob/release-polkadot-v1.3.0/substrate/frame/pallet-xcm/src/lib.rs#L192[{github-icon},role=heading-link]
+
+* Pallet-specific origins:
+** `AdminOrigin` -- The origin that is allowed to call privileged operations on the XCM pallet. Type must implement trait `EnsureOrigin`.
+** `ExecuteXcmOrigin` -- Required origin for executing XCM messages, including the teleport functionality. If successful, it returns a `MultiLocation` which exists as an interior location within this chain's XCM context. Type must implement trait `EnsureOrigin`.
+** `SendXcmOrigin` -- Required origin for sending XCM messages. If successful, it resolves to `MultiLocation`. Type must implement trait `EnsureOrigin`.
+
+* Pallet-specific ID types:
+** `RemoteLockConsumerIdentifier` -- The ID type for local consumers of remote locks. The type must implement `Copy`.
+
+* Pallet-specific handlers:
+** `OnChargeTransaction` -- Handler for withdrawing, refunding and depositing the transaction fee. Type must implement the trait  `OnChargeTransaction<Self>`.
+** `Currency` -- Lockable currency used in the pallet. Type must implement the trait `LockableCurrency`.
+** `Weigher` -- Measures weight consumed by XCM local execution. Type must implement the trait `WeightBounds`.
+** XcmExecutor -- Implements XCM messages. Type must implement the trait `ExecuteXcm`.
```

Comment:
> Implements XCM messages

is a bit misleading imo. I prefer the original explanation here instead: `Something to execute an XCM message.`, this was less confusing at least for me.